src/components/Card.js

```diff
@@ -1,53 +1,85 @@
 export default class Card {
-  constructor(cardData, cardSelector, handleImageClick) {
+  constructor(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeClick
+  ) {
     this._name = cardData.name;
     this._link = cardData.link;
+    this._id = cardData._id;
+    this.setIsLiked = cardData.isLiked;
     this._cardSelector = cardSelector;
-    this._handleImageClick = handleImageClick;
+    this.handleImageClick = handleImageClick;
+    this.handleDeleteCard = handleDeleteCard;
+    this.handleLikeClick = handleLikeClick;
   }
 
   _setEventListeners() {
     this._cardElement
-      .querySelector(".card__like-button")
+      .querySelector(".card__image")
       .addEventListener("click", () => {
-        this._handleLikeIcon();
+        this.handleImageClick({ name: this._name, link: this._link });
       });
 
     this._cardElement
       .querySelector(".card__delete-button")
       .addEventListener("click", () => {
-        this._handleDeleteCard();
+        this.handleDeleteCard(this);
       });
 
     this._cardElement
-      .querySelector(".card__image")
+      .querySelector(".card__like-button")
       .addEventListener("click", () => {
-        this._handleImageClick({ name: this._name, link: this._link });
+        this.handleLikeClick(this);
+      });
+
+    this._cardElement
+      .querySelector(".card__like-button")
+      .addEventListener("click", () => {
+        this.handleLikeIcon();
```

Comment:
```js

    this._cardElement
      .querySelector(".card__like-button")
      .addEventListener("click", () => {
        this.handleLikeIcon();
      });
```

Please, delete it

* One remark hasn't been corrected https://github.com/Cole-Greenwald/se_project_aroundtheus/pull/5#discussion_r1806245877

Please, delete the 2nd like handler. You have the correct one in line 35